docs/user/tutorial.rst

```diff
@@ -27,9 +26,47 @@ Now run the ``mnist.py`` example:
 
   python mnist.py
 
-XXX Explain output
+You will get the following output:
+
+.. code-block::text
+
+  Loading data...
+  Building model and compiling functions...
+/usr/local/lib/python2.7/dist-packages/Lasagne-0.1dev-py2.7.egg/lasagne/layers/helper.py:52: UserWarning: get_all_layers() has been changed to return layers in topological order. The former implementation is still available as get_all_layers_old(), but will be removed before the first release of Lasagne. To ignore this warning, use `  warnings.filterwarnings('ignore', '.*topo.*')`.
+    warnings.warn("get_all_layers() has been changed to return layers in "
+  Starting training...
+  Epoch 1 of 500 took 58.438s
+    training loss:        1.360644
+    validation loss:      0.467446
+    validation accuracy:  87.55 %%
+  Epoch 2 of 500 took 58.442s
+    training loss:        0.597908
+    validation loss:      0.330508
+    validation accuracy:  90.62 %%
+  Epoch 3 of 500 took 58.893s
+    training loss:        0.467016
+    validation loss:      0.278081
+    validation accuracy:  91.92 %%
+  Epoch 4 of 500 took 58.037s
+    training loss:        0.406298
+    validation loss:      0.248938
+    validation accuracy:  92.76 %%
+
+
+
+
 
 Understand the MNIST example
 ============================
 
-XXX Explain what's going on in the example
+The example starts with loading the MNIST dataset as train-, test- and
+validation set in line 206. It continues with automatically building a model
+with two hidden layers. It automatically detects the necessary number of input
+neurons (= number of features, 28*28=784 pixels in case of MNIST) and the
+necessary number of output neurons (= number of classes, 10 in case of MNIST).
+
+Then it automatically trains and evaluates the feed forward neural network
+with mini-batch gradient descent.
+
+A detailed explanation of how the training of feed forward neural networks
+works can be found in "Machine Learning" by  Tom Mitchell.
```

Comment:
Reading the full page again, I think it's misplaced here. In the introduction, you already state that "If you want to get detailed information on how neural networks work [...] read [...]". I think it's a good idea to list such prerequisites in the beginning. We don't aim for a tutorial explaining neural networks here, we just want to explain how Lasagne can help to build and train neural networks compared to using Theano directly.

Regarding the text book reference, it can be considered one of the standard text books, yes. I'm not sure if it's *the* standard text book, that's why I'm hesitating to refer to a single book.

What about removing it here and amending the sentence in the introduction:
[...] then you should read the Deeplearning Tutorial, possibly along with an online course for more theoretical background (["Neural Networks and Deep Learning" by Michael Nielsen](http://neuralnetworksanddeeplearning.com/), ["Convolutional Neural Networks for Visual Recognition" by Andrej Karpathy et al.](http://cs231n.github.io/)) or a standard text book such as "Machine Learning" by Tom Mitchell.